packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -319,7 +380,7 @@ export class ProfilesUtils {
         let fileContent: string;
         const settingsFile = path.join(getZoweDir(), "settings", "imperative.json");
         try {
-            fd = fs.openSync(settingsFile, "r+");
+            fd = fs.openSync(settingsFile, "w+");
```

Comment:
I think we want `r+` here to avoid overwriting the contents of the file if it already exists.

Reply:
w+ seems to solve our current existing issue with this function which creates this broken imperative.json when loading a plugin of a custom credential mananger and reloading Zowe Explorer, for example:
```JSON
{
    overrides: {
             credentialManager: "my-credential-manager"
    }
}
}
}
```

Reply:
I think the issue with a broken imperative.json was fixed in #2188:
> * When ZE updates the contents of `imperative.json` it should remove all old contents from the file even if the new contents are shorter.

If the `w+` works, perhaps we can simplify the code to remove the `catch` block? 😋